src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1836,13 +1836,17 @@ format layout_optimizer::get_preferred_format(program_node& node) {
         if (use_onednn_impls) {
             expected = node.get_preferred_output_fmt();
         }
-
-        if (!allow_new_shape_infer && node.is_type<fully_connected>()) {
-            auto& fc_node = node.as<fully_connected>();
-            auto input_layout = fc_node.get_input_layout();
-            if (input_layout.format.dimension() > 4) {
-                expected = format::bfyx;
-                node.set_preferred_input_fmt(0, format::bfyx);
+        if (node.is_type<fully_connected>()) {
+            if (allow_new_shape_infer) {
+                expected = format::get_default_format(node.get_input_layout(0).get_rank());
+                node.set_preferred_input_fmt(0, expected);
```

Comment:
Is it complicated to make a functional test case for such pattern? If not, I think it would be better to implement one..

Reply:
Added functional test.

Reply:
Could you please add a comment noting that plain input format is enforced because there is no available shape agnostic kernel supporting blocked format for now? Once we enable shape agnostic kernel for fsv->bfyx fc optimized kernel (i8), then we could relax the condition..

Reply:
Added the comment.